src/NHibernate.Test/LazyProperty/LazyPropertyFixture.cs

```diff
@@ -156,8 +167,14 @@ public void CanUpdateNonLazyWithoutLoadingLazyProperty()
 		}
 
 		[Test]
-		public void CanMergeWithLazyProperty()
+		public void CanMergeTransientWithLazyProperty()
 		{
+			using (ISession s = OpenSession())
+			{
+				var book = s.Get<Book>(2);
+				Assert.That(book, Is.Null);
```

Comment:
The test fails when merging a not yet persisted entity. Since we use assigned ids, we must ensure that any future changes will not cause this test to start merging an already persisted entity.